src/Jab/ServiceProviderBuilder.cs

```diff
@@ -123,7 +124,12 @@ internal class ServiceProviderBuilder
     public ServiceProviderBuilder(GeneratorContext context)
     {
         _context = context;
-        _knownTypes = new KnownTypes(context.Compilation, context.Compilation.Assembly);
+
+        var assemblySymbol =
```

Comment:
I am worried that we have two ways of detecting Jab.Attributes being referenced. Is it possible to unify to only looking at Compilation.SourceModule.ReferencedAssemblySymbols?

Reply:
It will need to change the parameters of `ProcessModule` method

Reply:
Oh, I mean the fact that we have a JAB_ATTRIBUTES_REFERENCED variable that controls Attributes.cs injection but KnownTypes work differently.

Can we switch to using the same logic when adding Attributes.cs?

Reply:
Just looked, I guess not. RegisterPostInitializationOutput doesn't have enough information it seems.

Reply:
I'm not sure if I understood

Reply:
Ignore me, what you have is good :)